src/es-router.js

```diff
@@ -229,34 +261,61 @@ class EsRouter {
     }, []);
   }
 
-  //actual routing function
-  path(route, isQueryParam) {
+  /**
+   * Build a new url for path changes
+   * We assume the route has already been verified as a predefined route here (or not)
+   *
+   * @param  {string} newPath - input path, e.g. /account/login
+   * @return {string}         - output ready for history push, e.g. /base/account/login?foo=bar
+   */
+  buildNewUrl(newPath) {
+    const paramArray = this.createParamString(this.queryParams);
+    const paramArrayString = paramArray.length ? `?${paramArray.join('&')}` : '';
+    // make sure the new url starts with a slash
+    const newUrlBase = this.base.match(/^\//) ? this.base : `/${this.base}`;
+    const newUrl = (`${newUrlBase}${newPath}${paramArrayString}`)
+      .replace(/\/{2,}/g, '/'); // dedup consecutive slashes
+    return newUrl;
+  }
+
+  /**
+   * Actual routing function
+   *
+   * @param  {string}  route        - New path to naviate to. Absolute path, not including base
```

Comment:
I dig your use of "nit" @chrisrng.

Had to look it up to confirm it meant what I thought it did and found it in the chromium glossary.
https://www.chromium.org/glossary

short for "nitpick"; refers to a trivial suggestion such as style issues